docs/how_to_publish_to_cmr_with_earthdata_varinfo.ipynb

```diff
@@ -0,0 +1,413 @@
+   "metadata": {},
+   "source": [
+    "Import `generate_collection_umm_var` from `varinfo.generate_umm_var`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from varinfo.generate_umm_var import generate_collection_umm_var"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`generate_collection_umm_var` will:\n",
+    "\n",
+    "* Download the most recent granule for **GLDAS_NOAH10_3H**\n",
+    "* Generate the UMM-Var records for this granule\n",
+    "* Publish these records to CMR if `publish=True` and return a list of published variable concept-ids\n",
+    "(e.g. `['V1259971755-EEDTEST', 'V1259971757-EEDTEST' ...]`)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "generate_collection_umm_var(collection_concept_id=collection_concept_id_gldas,\n",
+    "                            auth_header=auth_header, publish=True)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Publishing and creating UMM-Var entries for **M2I1NXASM** with:\n",
+    "\n",
+    "* `varinfo.cmr_search`: queries CMR for a granule download link and downloads granules locally\n",
+    "* `VarInfoFromNetCDF4`: varinfo parent class that represents the contents of a granule\n",
+    "* `varinfo.umm_var`: contains functions for creating and publishing UMM-Var records to CMR\n",
+    "* `CMR_UAT` is a string constant (e.g. https://cmr.uat.earthdata.nasa.gov/search/) of a CMR environment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from cmr import CMR_UAT\n",
+    "\n",
+    "from varinfo.cmr_search import (get_granules, get_granule_link, \n",
+    "                                download_granule)\n",
+    "\n",
+    "from varinfo import VarInfoFromNetCDF4\n",
+    "\n",
+    "from varinfo.umm_var import (get_all_umm_var, get_umm_var, publish_all_umm_var,\n",
+    "                             publish_umm_var)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Update the `collection_concept_id` to the **M2I1NXASM** concept-id for the EEDTEST provider\n",
+    "* This can be updated to any concept-id for any provider"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "collection_concept_id_merra = 'C1256535511-EEDTEST'"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Get the granule record and granule download URL with `get_granules` and `get_granule_link`\n",
+    "\n",
+    "* `get_granules`: queries `CMR_OPS` for a UMM-G record (granule record) given a collection or granule concept-id\n",
```

Comment:
Nit: Doesn't it query `CMR_UAT` rather than `CMR_OPS`? (Although totally worth saying something about "if you want to query production, change this to `CMR_OPS`...")

Reply:
`cmr_search.get_granules` queries `CMR_OPS` by default. Here is the [line](https://github.com/nasa/earthdata-varinfo/blob/main/varinfo/cmr_search.py#L25C1-L25C1)

Reply:
Totally fair, but in this notebook we are setting the value to `CMR_UAT` in the cell below. (Maybe the extra text should be to clarify there's a default value of `CMR_OPS`)

Reply:
Oh right we are doing that haha!

Reply:
Since it's just a search query I am going to remove that line in `get_granules` (`cmr_env=CMR_UAT` ) and just leave it as the default behavior which is OPS and leave something in the comments about being able to change it from the default, if that's cool?

Reply:
* [ ] Add text about being able to query different environments

Reply:
I think the reason the `cmr_env=CMR_UAT` was included was to avoid someone accidentally just running all the cells and updating production metadata records, which are end-user facing. But either way works, so long as it's clear what the user has to do. (Maybe we need a `sudo`/Spiderman disclaimer at the top of the notebook about great power/responsibility?)

Reply:
For all of the cells with `publish_all_umm_var` and `publish_umm_var` I have the environment set to `CMR_UAT` so in the case that a user was to run through all the cells at once it won't update production metadata records